src/containers/Activity/Activity.js

```diff
@@ -50,11 +57,9 @@ export class Activity extends React.PureComponent {
       );
     }
   }
-  componentWillReceiveProps(nextProps){
-    if(
-      !isEqual(nextProps.pathProblem, this.props.pathProblem)
-    ){
-    this.onProblemChange({});
+  componentWillReceiveProps(nextProps) {
```

Comment:
@Brainenjii do you think `componentWillReceiveProps()` in our code base should all be replaced?

Reply:
Yeah, I think we should consider avoiding using deprecated (and maybe also unsafe) features